.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseSetParser.cs

```diff
@@ -32,61 +33,78 @@ public DateTimeParseResult Parse(ExtractResult er, DateObject refDate)
 
             if (er.Type.Equals(ParserName, StringComparison.Ordinal))
             {
-                var innerResult = ParseEachUnit(er.Text);
+                var innerResult = ParseEachUnit(er.Text.Trim());
 
                 if (!innerResult.Success)
                 {
-                    innerResult = ParseEachDuration(er.Text, refDate);
+                    innerResult = ParseEachDuration(er.Text.Trim(), refDate);
                 }
 
                 if (!innerResult.Success)
                 {
-                    innerResult = ParserTimeEveryday(er.Text, refDate);
+                    innerResult = ParserTimeEveryday(er.Text.Trim(), refDate);
                 }
 
                 // NOTE: Do not change the order of the following calls, due to type precedence
                 // datetimeperiod > dateperiod > timeperiod > datetime > date > time
                 if (!innerResult.Success)
                 {
-                    innerResult = ParseEach(config.DateTimePeriodExtractor, config.DateTimePeriodParser, er.Text, refDate);
+                    innerResult = ParseEach(config.DateTimePeriodExtractor, config.DateTimePeriodParser, er.Text.Trim(), refDate);
                 }
 
                 if (!innerResult.Success)
                 {
-                    innerResult = ParseEach(config.DatePeriodExtractor, config.DatePeriodParser, er.Text, refDate);
+                    innerResult = ParseEach(config.DatePeriodExtractor, config.DatePeriodParser, er.Text.Trim(), refDate);
                 }
 
                 if (!innerResult.Success)
                 {
-                    innerResult = ParseEach(config.TimePeriodExtractor, config.TimePeriodParser, er.Text, refDate);
+                    innerResult = ParseEach(config.TimePeriodExtractor, config.TimePeriodParser, er.Text.Trim(), refDate);
                 }
 
                 if (!innerResult.Success)
                 {
-                    innerResult = ParseEach(config.DateTimeExtractor, config.DateTimeParser, er.Text, refDate);
+                    innerResult = ParseEach(config.DateTimeExtractor, config.DateTimeParser, er.Text.Trim(), refDate);
                 }
 
                 if (!innerResult.Success)
                 {
-                    innerResult = ParseEach(config.DateExtractor, config.DateParser, er.Text, refDate);
+                    innerResult = ParseEach(config.DateExtractor, config.DateParser, er.Text.Trim(), refDate);
                 }
 
                 if (!innerResult.Success)
                 {
-                    innerResult = ParseEach(config.TimeExtractor, config.TimeParser, er.Text, refDate);
+                    innerResult = ParseEach(config.TimeExtractor, config.TimeParser, er.Text.Trim(), refDate);
+                }
+
+                if (!innerResult.Success)
+                {
+                    innerResult = ParserDayEveryweek(er.Text.Trim(), refDate);
```

Comment:
ParserDayEveryWeek?

Reply:
ParserDayEveryWeek is used for handling example such as 19th of every month. Added comment describing it.